components/desktop/lightdm/patches/20-sdtlogin.patch

```diff
@@ -0,0 +1,188 @@
++                  pipe, g_strerror(errno));
++          return;
++     }
++     fp = fdopen (fd, "w");
++     if (fp == NULL) {
++          close (fd);
++          g_warning ("solaris_xserver_cred: could not fdopen"
++                  " %s: %s\n", pipe, g_strerror(errno));
++          return;
++     }
++    }
++
++    snprintf (info, sizeof(info), "GID=\"%d\"; ", gid);
++    fputs (info, fp);
++    g_debug ("solaris_xserver_cred: %s\n", info);
++
++    if (initgroups (username, gid) == -1) {
++         ngroups = 0;
++    } else {
++         ngroups = getgroups (NGROUPS_UMAX, groups);
++    }
++
++    for (i=0; i < ngroups; i++) {
++         snprintf (info, sizeof(info), "G_LIST_ID=\"%u\" ", groups[i]);
++         fputs (info, fp);
++         g_debug ("solaris_xserver_cred: %s\n", info);
++    }
++
++    if (ngroups > 0) {
++         fputc (';', fp);
++    }
++
++    snprintf (info, sizeof(info), " HOME=\"%s\" ", home_dir);
++    fputs (info, fp);
++    g_debug ("solaris_xserver_cred: %s\n", info);
++
++    snprintf (info, sizeof(info), " UID=\"%d\" EOF=\"\";",
++        uid);
++    fputs (info, fp);
++    g_debug ("solaris_xserver_cred: %s\n", info);
++
++    /*
++     * Handshake with server. Make sure it read the pipe.
++     *
++     * Do not close file descriptor, but leave it open for further use.
++     */
++    fflush (fp);
++
++    return;
++}
++
+ static gchar *
+ read_string (void)
+ {
+@@ -854,6 +964,13 @@
+     uid = user_get_uid (user);
+     gid = user_get_gid (user);
+     home_directory = user_get_home_directory (user);
++
++#ifdef __sun
++    if (xdisplay) { /* Should always be true */
++        solaris_xserver_cred (username, uid, gid, home_directory, xdisplay);
++    }
++#endif
++
+     child_pid = fork ();
+     if (child_pid == 0)
+     {
+--- lightdm-1.19.3/common/configuration.h.1	2017-10-25 08:43:04.689998057 +0000
++++ lightdm-1.19.3/common/configuration.h	2017-10-25 08:44:56.582620720 +0000
+@@ -12,6 +12,15 @@
+ #ifndef CONFIGURATION_H_
+ #define CONFIGURATION_H_
+ 
++/* 
++ * Perhaps, it's not the best place for it, but it's the only common header 
++ * of lightdm.c and sesion-child.c
++ */
++#ifdef __sun
++#define DT_DIR "/var/dt"
++#define SDTLOGIN_DIR "/var/dt/sdtlogin"
++#endif
++
+ #include <glib-object.h>
+ 
+ G_BEGIN_DECLS
+--- lightdm-1.19.3/src/lightdm.c.~4~	2017-10-25 09:12:24.672787804 +0000
++++ lightdm-1.19.3/src/lightdm.c	2017-10-25 11:05:49.884502470 +0000
+@@ -16,6 +16,7 @@
+ #include <sys/stat.h>
+ #include <glib.h>
+ #include <glib/gi18n.h>
++#include <glib/gstdio.h>
+ #include <unistd.h>
+ #include <fcntl.h>
+ #include <sys/stat.h>
+@@ -1529,6 +1530,23 @@
+         g_warning ("Failed to make cache directory %s: %s", dir, strerror (errno));
+     g_free (dir);
+ 
++#ifdef __sun
++    {
++        struct stat statbuf;
++        int r;
++
++        r = stat (DT_DIR, &statbuf);
++        if (r < 0) {
++           g_mkdir (DT_DIR, 0755);
++        }
++
```

Comment:
Original patch does 
```
g_rmdir(SDTLOGIN_DIR);
g_mkdir(SDTLOGIN_DIR,0700); 
```
But this will not work if this directory contains any pipes. So we just check that directory exists, and if it doesn't, create it.

Reply:
OK , thanks for the explanation.